src/ComparisonWithNumbers.js

```diff
@@ -0,0 +1,127 @@
+
+class ComparisonWithNumbers {
+  constructor() {
+    this.userMoney;
+    this.userNumbers;
+    this.numberOfMatchingBalls;
+    this.index = Object.keys(PRIZE_MONEY).length;
+  }
+
+  compareWithNumbers(userNumbers, LotteryTickets, userMoney) {
+    this.userMoney = userMoney;
+    this.userNumbers = userNumbers;
+
+    let index = LotteryTickets.length;
+    while (index--) {
+      LotteryTickets[index] = LotteryTickets[index].filter((num) =>
+        userNumbers.includes(num)
+      );
+    }
+
+    return this.findNumberOfMatchingBalls(LotteryTickets);
+  }
+
+  findNumberOfMatchingBalls(LotteryTickets) {
+    this.numberOfMatchingBalls = LotteryTickets.map((ticket) => ticket.length);
+
+    this.acceptBonusNumber();
+  }
+
+  acceptBonusNumber() {
+    CONSOLE_UTIL.readLine(INPUT_BONUS_NUMBERS_MESSAGE, (userBonus) => {
+      this.validateBonusNumber(userBonus);
+    });
+  }
+
+  validateBonusNumber(userBonus) {
+    if (!FROM_ONE_TO_FORTYFIVE.includes(Number(userBonus))) {
+      throw new Error(NOT_BETWEEN_ONE_AND_FORTYFIVE);
+    }
+
+    return this.makeBonusNumber(userBonus);
+  }
+
+  makeBonusNumber(userBonus) {
+    const lottoBonus = RANDOM_UTIL.pickNumberInRange(1, 45);
+
+    this.isSameBonusNumber(userBonus, lottoBonus);
+  }
+
+  isSameBonusNumber(userBonus, lottoBonus) {
+    const isSameBonus = userBonus === lottoBonus;
+
+    this.findNumberOfWinningTickets(isSameBonus);
+  }
+
+  findNumberOfWinningTickets(isSameBonus) {
+    const places = {};
+    places[5] = this.numberOfMatchingBalls.filter((num) => num === 3).length;
+    places[4] = this.numberOfMatchingBalls.filter((num) => num === 4).length;
+    places[3] = places[2] = 0;
+    this.compareWithBonus(places, isSameBonus);
+    places[1] = this.numberOfMatchingBalls.filter((num) => num === 6).length;
+
+    this.printResult(places);
+  }
+
+  compareWithBonus(places, isSameBonus) {
+    const SecondOrThird = this.numberOfMatchingBalls.filter(
+      (num) => num === 5
+    ).length;
+    if (SecondOrThird) {
+      isSameBonus ? (places[2] = SecondOrThird) : (places[3] = SecondOrThird);
+    }
+  }
+
+  printResult(numberOfWinningTickets) {
+    CONSOLE_UTIL.print(RESULT_MESSAGE);
+
+    let { index } = this;
+    for (index; index > 0; index--) {
+      CONSOLE_UTIL.print(
+        `${PRIZE_MESSAGES[index]} - ${numberOfWinningTickets[index]}개`
+      );
+    }
+
+    this.calculateRateOfReturn(numberOfWinningTickets);
+    this.endGame();
+  }
+
+  calculateRateOfReturn(numberOfWinningTickets) {
+    let { index } = this;
+    let totalPrize = 0;
+    for (index; index > 0; index--) {
+      totalPrize += numberOfWinningTickets[index] * PRIZE_MONEY[index];
+    }
+
+    return this.printRateOfReturn(totalPrize);
+  }
+
+  printRateOfReturn(totalPrize) {
+    const rate = ((totalPrize / this.userMoney) * 100).toFixed(1);
+    const addCommasToNumbers = String(rate).replace(
+      /\B(?=(\d{3})+(?!\d))/g,
+      ","
+    );
```

Comment:
세자리 수마다 콤마찍는 방법 중 이런 방법도 있어요😎
```suggestion
    const addCommasToNumbers = Number(rate).toLocaleString();
```